core/trino-spi/src/main/java/io/trino/spi/block/VariableWidthBlockEncoding.java

```diff
@@ -19,12 +19,10 @@
 import io.airlift.slice.Slices;
 import jakarta.annotation.Nullable;
 
-import java.util.Arrays;
-
 import static io.trino.spi.block.EncoderUtil.decodeNullBits;
 import static io.trino.spi.block.EncoderUtil.encodeNullsAsBits;
-import static java.lang.String.format;
-import static java.util.Objects.checkFromIndexSize;
+import static io.trino.spi.block.EncoderUtil.readOffsetsWithNullsCompacted;
+import static io.trino.spi.block.EncoderUtil.writeOffsetsWithNullsCompacted;
```

Comment:
Why do these methods belong in a utils class ?
These seem specific to VariableWidthBlockEncoding and have only one usage, so seems more appropriate to leave it in this class